torch/_subclasses/fake_tensor.py

```diff
@@ -531,8 +531,21 @@ class FakeTensorConfig:
 # Making this a descriptor may seem overly fancy, but actually it's the most
 # convenient way to make sure we have access to FakeTensor during access,
 # which is required for testing version counter and epoch validity
-class UnbackedMemoDescriptor:
+class SymIntMemoDescriptor:
     _name: str
+    _is_unbacked: bool
+    # This flag disables version counting. We use this for NJT, where
+    # memo of the nested int on the fake offsets should not be invalidated
+    # even if the offsets is mutated. This is true given that each nested
+    # is not shared across multiple distinct tensors.
+    _no_vc: bool
+
+    def __init__(
+        self, *, is_unbacked: Optional[bool] = None, no_vc: bool = False
+    ) -> None:
+        assert is_unbacked is not None
+        self._is_unbacked = is_unbacked
+        self._no_vc = no_vc
```

Comment:
no_vc is the operational description here, but it might be more clear if it was something like `preserve_across_mutation: bool = False`

Reply:
Or even better, just have a `nested_int` flag that controls both the is unbacked and no vc simultaneously, since that's the only use case right now. Not a big deal though.